Automattic/vip-go-mu-plugins
# Don't try to update the index settings if the setting key is not in the list of INDEX_SETTINGS_HEALTH_AUTO_HEAL_KEYS

## Description

Don't try to correct index settings if an inconsistent value is not a part of INDEX_SETTINGS_HEALTH_AUTO_HEAL_KEYS list.

## Changelog Description

### Plugin Updated: Search

Improve internal index settings monitoring job to skip trying to fix inconsistencies in non-auto-healable settings.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

If you have Elasticsearch Head Chrome extension installed and configured:
1. Add a dynamic setting in the list of monitored keys, E.g. `index.max_shingle_size` but don't add to the list of auto-healable keys
2. Connect to the server in ES Head:
```PUT _settings

{
  "index": {
    "max_result_window": 20000,
    "max_shingle_size": 10
  }
}
```

3. Add `var_dump for `$result['diff']` and `$diff`. 
4. Verify that the latter doesn't contain `max_shingle_size`
